velox/serializers/UnsafeRowSerializer.cpp

```diff
@@ -112,11 +112,31 @@ void UnsafeRowVectorSerde::deserialize(
     RowVectorPtr* result,
     const Options* /* options */) {
   std::vector<std::optional<std::string_view>> serializedRows;
+  std::vector<std::string> concatenatedRows;
+
   while (!source->atEnd()) {
     // First read row size in big endian order.
     auto rowSize =
         folly::Endian::big(source->read<UnsafeRowVectorSerializer::TRowSize>());
     auto row = source->nextView(rowSize);
+
+    // If we couldn't read the entire row at once, we need to concatenate it
```

Comment:
Do we expect this to happen very rarely?

Reply:
Sort of. It happens when a record was serialized in between two Pages, so parts of it were in different ranges.